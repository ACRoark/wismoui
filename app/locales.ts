export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
