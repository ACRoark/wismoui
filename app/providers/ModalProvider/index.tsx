/**
 *
 * ModalProvider
 *
 */
import React, { useState } from 'react';

import IModalConfig from 'types/IModalConfig';
import IModalState from 'types/IModalState';

import Modal from './Modal';

import 'react-responsive-modal/styles.css';

interface IModalProviderProps {
  children?: React.ReactNode;
}

const ModalContext = React.createContext<IModalState | null>(null);

const ModalConsumer = ModalContext.Consumer;

const ModalProvider: React.FC<IModalProviderProps> = (props: IModalProviderProps): React.ReactElement => {
  const { children } = props;

  const [modalConfig, setModalConfig] = useState<IModalConfig | null>(null);

  const showModal = (config: IModalConfig): void => {
    setModalConfig(config);
  };

  const hideModal = (): void => {
    setModalConfig(null);
  };

  return (
    <ModalContext.Provider
      value={{
        hideModal,
        showModal,
      }}
    >
      {React.Children.only(children)}
      {modalConfig && <Modal onClose={hideModal} config={modalConfig} />}
    </ModalContext.Provider>
  );
};

export { ModalConsumer, ModalContext, ModalProvider };
