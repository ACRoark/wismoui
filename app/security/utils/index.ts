import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
