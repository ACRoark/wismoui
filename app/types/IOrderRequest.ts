import IOrderRequestBase from './IOrderRequestBase';
import IOrderRequestStatusUpdate from './IOrderRequestStatusUpdate';
import IServiceOrderEvent from './IServiceOrderEvent';
import IVerificationEvent from './IVerificationEvent';

interface IOrderRequest extends IOrderRequestBase {
  contactPhoneNumber: string;
  orderRequestStatusUpdates: IOrderRequestStatusUpdate[];
  serviceOrderEvents: IServiceOrderEvent[];
  verificationEvents: IVerificationEvent[];
}

export default IOrderRequest;
