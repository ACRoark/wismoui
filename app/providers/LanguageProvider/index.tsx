import React, { useState } from 'react';
import { IntlProvider } from 'react-intl';

import ILanguageState from 'types/ILanguageState';

// Import i18n messages
import { translationMessages } from '../../i18n';

import { DEFAULT_LOCALE, SUPPORTED_LOCALES } from '../../locales';

interface ILanguageProviderProps {
  children?: React.ReactNode;
  locale?: string;
}

const LanguageContext = React.createContext<ILanguageState | null>(null);

const LanguageConsumer = LanguageContext.Consumer;

const LanguageProvider: React.FC<ILanguageProviderProps> = (props: ILanguageProviderProps): React.ReactElement => {
  const { children, locale = DEFAULT_LOCALE } = props;

  const [language, setLanguage] = useState<string>(locale);

  // TODO: Remove the slice operation if regional variants are being supported
  //       E.g. en-CA, en-GB, en-US instead of just en
  const messages = translationMessages[language.length > 2 ? language.slice(0, 2) : language];

  return (
    <LanguageContext.Provider
      value={{
        changeLanguage: setLanguage,
        language,
        supportedLanguages: SUPPORTED_LOCALES,
      }}
    >
      <IntlProvider key={locale} locale={locale} messages={messages}>
        {React.Children.only(children)}
      </IntlProvider>
    </LanguageContext.Provider>
  );
};

export { LanguageConsumer, LanguageContext, LanguageProvider };
