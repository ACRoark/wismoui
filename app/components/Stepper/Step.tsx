import React, { FC, ReactElement } from 'react';
import { Description } from 'types';

import StepIcon from './StepIcon';
import { StepperStatus } from './types';

interface IStepProps {
  description: Description;
  status: StepperStatus;
  title: string | ReactElement;
}

const Step: FC<IStepProps> = (props: IStepProps): ReactElement => {
  const { description, status, title } = props;

  return (
    <div className="dte-wismo-step">
      <div className={`container ${status}`}>
        <div className="tail">
          <div className="half-1" />
          <div className="half-2" />
        </div>
        <div className="icon-wrapper">
          <StepIcon status={status} />
        </div>
        <div className="content">
          <div className="title">{title}</div>
          <div className="description">{description}</div>
        </div>
      </div>
    </div>
  );
};

export default Step;
