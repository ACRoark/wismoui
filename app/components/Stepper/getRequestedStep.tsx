import React from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import IOrderRequest from 'types/IOrderRequest';
import messages from './messages';
import { IStepDescription, StepperStatus } from './types';

const getDescription = (orderCreatedDate: string, orderRequest: IOrderRequest): React.ReactElement => {
  const orderCreated = <DateStamp format="short" value={orderCreatedDate} />;

  if (orderRequest.orderRequestStatusUpdates.length === 0) {
    return (
      <span>
        <div>{orderCreated}</div>
        <FormattedMessage {...messages.newOrder} />
      </span>
    );
  }

  return orderCreated;
};

const getRequestedStep = (
  orderCreatedDate: string,
  serviceRequest?: IOrderRequest,
): IStepDescription => {
  if (serviceRequest && serviceRequest.orderRequestStatusUpdates.length === 0) {
    return {
      description: getDescription(orderCreatedDate, serviceRequest),
      key: 'Request Received',
      status: StepperStatus.Completed,
      title: <FormattedMessage {...messages.request} />,
    };
  }

  return {
      description: <DateStamp format="short" value={orderCreatedDate} />,
      key: 'Request Received',
      status: StepperStatus.Completed,
      title: <FormattedMessage {...messages.request} />,
  };
};

export default getRequestedStep;
