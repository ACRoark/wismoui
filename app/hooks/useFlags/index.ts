import { useSelector } from 'react-redux';

import { IApplicationRootState, IFeatureFlags } from 'types';

const useFlags = (): IFeatureFlags => useSelector((state: IApplicationRootState): IFeatureFlags => state.config.flags);

export default useFlags;
