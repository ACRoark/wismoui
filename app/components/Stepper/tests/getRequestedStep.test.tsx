import React from 'react';
import { Provider } from 'react-redux';

import createFakeOrder from 'components/createFakeOrder';
import useFlags from 'hooks/useFlags';
import { ModalProvider } from 'providers/ModalProvider';
import { createMockStore, createSnapshotWithIntl } from 'testing/utils';
import IOrderRequest from 'types/IOrderRequest';
import Step from '../Step';

import getRequestedStep from '../getRequestedStep';

import { SUPPORTED_LOCALES } from 'locales';

jest.mock('hooks/useFlags');

const fakeOrder = createFakeOrder('MI12345678');
const fakeServiceRequest = fakeOrder.orderRequests[0];
const newServiceRequest = {
  ...fakeServiceRequest,
  orderRequestStatusUpdates: [],
};
const mockFlags = useFlags as jest.MockedFunction<typeof useFlags>;
const store = createMockStore();

const assertRequestedStepMatchesSnapshot = (serviceRequest: IOrderRequest, language: string): void => {
  mockFlags.mockReturnValue({
    canChangeLanguage: true,
    developerMode: true,
  });

  const stepDesc = getRequestedStep('09-22-2020', serviceRequest);

  const tree = createSnapshotWithIntl(
    <Provider store={store}>
      <ModalProvider>
        <Step {...stepDesc} />
      </ModalProvider>
    </Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('getRequestedStep', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  languages.forEach((language: string): void => {
    it(`should return the date the order was created when language is ${language}`, (): void => {
      assertRequestedStepMatchesSnapshot(fakeServiceRequest, language);
    });

    it(`should return the date the order was created and a 'new order' message when language is ${language}`, (): void => {
      assertRequestedStepMatchesSnapshot(newServiceRequest, language);
    });

    it(`should return the date the order was created when language is ${language}, and orderRequestStatusUpdates array is not empty`, (): void => {
      assertRequestedStepMatchesSnapshot(fakeServiceRequest, language);
    });
  });
});
