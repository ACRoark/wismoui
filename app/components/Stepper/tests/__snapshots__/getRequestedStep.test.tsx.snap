// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRequestedStep should return the date the order was created and a 'new order' message when language is en 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Order Received
      </div>
      <div
        className="description"
      >
        <span>
          <div>
            09/22/20
          </div>
          Please allow up to 30 minutes for us to process your order. Thank you for your patience.
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`getRequestedStep should return the date the order was created and a 'new order' message when language is es 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Orden recibida
      </div>
      <div
        className="description"
      >
        <span>
          <div>
            22/09/20
          </div>
          Por favor, espere hasta 30 minutos para que procesemos su pedido. Gracias por su paciencia.
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`getRequestedStep should return the date the order was created when language is en 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Order Received
      </div>
      <div
        className="description"
      >
        09/22/20
      </div>
    </div>
  </div>
</div>
`;

exports[`getRequestedStep should return the date the order was created when language is en, and orderRequestStatusUpdates array is not empty 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Order Received
      </div>
      <div
        className="description"
      >
        09/22/20
      </div>
    </div>
  </div>
</div>
`;

exports[`getRequestedStep should return the date the order was created when language is es 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Orden recibida
      </div>
      <div
        className="description"
      >
        22/09/20
      </div>
    </div>
  </div>
</div>
`;

exports[`getRequestedStep should return the date the order was created when language is es, and orderRequestStatusUpdates array is not empty 1`] = `
<div
  className="dte-wismo-step"
>
  <div
    className="container completed"
  >
    <div
      className="tail"
    >
      <div
        className="half-1"
      />
      <div
        className="half-2"
      />
    </div>
    <div
      className="icon-wrapper"
    >
      <svg
        className="completed icon"
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
          fillRule="evenodd"
          stroke="currentColor"
          strokeWidth="1px"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <div
        className="title"
      >
        Orden recibida
      </div>
      <div
        className="description"
      >
        22/09/20
      </div>
    </div>
  </div>
</div>
`;
